chain/processor.go

```diff
@@ -138,21 +103,15 @@ func (b *StatelessBlock) Execute(
 
 			// Commit results to parent [TState]
 			tsv.Commit()
-
-			// Update key cache
-			if len(toCache) > 0 {
-				cacheLock.Lock()
-				for k := range toCache {
-					cache[k] = toCache[k]
-				}
-				cacheLock.Unlock()
-			}
 			return nil
 		})
 	}
 	if err := e.Wait(); err != nil {
 		return nil, nil, err
 	}
+	if err := f.Wait(); err != nil {
```

Comment:
With the changes that have been made, it should now be safe to put `f.Wait` before `e.Wait` again: https://github.com/ava-labs/hypersdk/pull/744#discussion_r1513342923